src/misc/debug.h

```diff
@@ -19,8 +21,20 @@ extern int debug_level;
 extern _Bool debug_logging;
 
 enum debug_levels { DP_ERROR, DP_WARN, DP_INFO, DP_DEBUG1, DP_DEBUG2, DP_DEBUG3, DP_DEBUG4, DP_TRACE, DP_ALL };
-extern void debug_printf(int level, const char* fmt, ...);
-extern void debug_vprintf(int level, const char* fmt, va_list ap);
+extern void debug_vprintf_trace(const char* func_name,
+				const char* file,
+				unsigned int line,
+				int level, const char* fmt, va_list ap);
+extern void debug_printf_trace(const char* func_name,
+			       const char* file,
+			       unsigned int line,
+			       int level, const char* fmt, ...);
+
+// To get the proper function name, file and line when spoofing UTrace
+#define debug_printf(level, ...)					\
```

Comment:
These redefinition should be optional depending on a preprocessor variable

Reply:
Agreed.

Reply:
Good idea.

Reply:
The same for debug_print_dims. Thank you!